File: testdata/kotlin/original.kt
// File header comment
package com.example

// Import comments
import kotlin.collections.*

/**
 * Class documentation comment
 * @property name The name property
 */
@Entity
@Suppress("UNCHECKED_CAST") // Suppress warning annotation
class Person(
    // Constructor parameter comment
    private val name: String,
    val age: Int // Age property
) {
    // Property comment
    private val id: Long = 0L
    
    /**
     * Function documentation 
     * @param greeting The greeting message
     * @return formatted greeting
     */
    @JvmStatic // Platform annotation
    fun greet(greeting: String): String {
        // Single line comment
        return "$greeting, $name!" // Inline comment
    }
    
    /* 
     * Multi-line comment block
     * with multiple lines
     */
    private fun validate() {
        // Validation logic
        if (name.isEmpty()) { // Check name
            throw IllegalArgumentException("Name cannot be empty") // Error message
        }
    }
}

// Companion object comment
@Deprecated("Use factory method instead") // Deprecation annotation
object PersonFactory {
    // Factory method comment
    fun createPerson(name: String, age: Int): Person {
        /* Block comment before return */
        return Person(name, age)
    }
} 

File: testdata/swift/original.swift
// Single line comment at the top
import Foundation

/// Documentation comment for the class
/// This is a multi-line documentation comment
class SwiftExample {
    // Property comment
    var name: String = "example" // Inline comment
    
    /* Multi-line comment
       spanning multiple lines
       with more details */
    private var count: Int = 0
    
    /**
     * JavaDoc style documentation comment
     * @param name The name parameter
     * @returns Void
     */
    func setName(_ name: String) {
        // Simple line comment
        self.name = name // Another inline comment
    }
    
    // TODO: Implement this method
    func increment() {
        /* Block comment */ count += 1
    }
    
    /// Single line doc comment
    func getValue() -> Int {
        return count // Return comment
    }
}

// Final comment at the end 